### PR TITLE
Gh6511 2 repair PaySelectionTrxType calculation

### PR DESCRIPTION
**Pull request by metas-rc**
https://github.com/metasfresh/metasfresh/issues/6511

### CODE REVIEW

File: backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionUpdater.java [hunk 1]
```diff
@@ -386,7 +386,7 @@ else if (getC_BP_Group_ID() > 0)
 	private String buildSelectSQL_MatchRequirement()
 	{
 		final String whereCreditTransferToVendor = " i.IsSOTrx='N' AND i.PaymentRule IN ('" + PaymentRule.DirectDeposit.getCode() + "','" + PaymentRule.OnCredit.getCode() + "') ";
-		final String whereDirectDebitFromCustomer = " i.IsSOTrx='Y' AND dt.DocBaseType!='ARC' AND i.PaymentRule IN ('" + PaymentRule.DirectDebit.getCode() + "') ";
+		final String whereDirectDebitFromCustomer = " i.IsSOTrx='Y' AND dt.DocBaseType!='ARC' AND i.PaymentRule IN ('" + PaymentRule.DirectDebit.getCode() + "','" + PaymentRule.OnCredit.getCode() + "') ";
```
Comment (teosarca) [inline backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionUpdater.java:389]: pls change `!='ARC'` to `='ARI'`.

Reason: avoid future doc base types to be matched by this rule.

File: backend/de.metas.payment.sepa/base/src/main/sql/postgresql/ddl/views/C_Invoice_PaySelectionTrxType_V.sql [hunk 2]
```diff
@@ -4,8 +4,8 @@ CREATE OR REPLACE VIEW C_Invoice_PaySelectionTrxType_V AS
 SELECT C_Invoice_ID,
 	CASE 
 		WHEN /* CREDIT_TRANSFER_TO_VENDOR - "OUT" */ i.IsSOTrx='N' AND i.PaymentRule IN ('T'/*DirectDeposit*/, 'P'/*OnCredit*/) AND i.DocStatus IN ('CO', 'CL') THEN 'CT'
-		WHEN /* DIRECT_DEBIT_FROM_CUSTOMER - "CDD" */ i.IsSOTrx='Y' AND dt.DocBaseType = 'ARC' AND i.PaymentRule IN ('D'/*DirectDebit*/) AND i.DocStatus IN ('CO', 'CL') THEN 'DD'
-		WHEN /* CREDIT_TRANSFER_TO_CUSTOMER - "CRE" */ i.IsSOTrx='Y' AND dt.DocBaseType = 'ARC' AND i.PaymentRule IN ('T'/*DirectDeposit*/) AND i.DocStatus IN ('CO', 'CL') THEN 'CT'
+		WHEN /* DIRECT_DEBIT_FROM_CUSTOMER - "CDD" */ i.IsSOTrx='Y' AND dt.DocBaseType != 'ARC' AND i.PaymentRule IN ('D'/*DirectDebit*/, 'P'/*OnCredit*/) AND i.DocStatus IN ('CO', 'CL') THEN 'DD'
```
Comment (teosarca) [inline backend/de.metas.payment.sepa/base/src/main/sql/postgresql/ddl/views/C_Invoice_PaySelectionTrxType_V.sql:7]: pls change `!='ARC'` to `='ARI'`.

Reason: avoid future doc base types to be matched by this rule.

File: backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionUpdater.java (hunk 1 above)
Comment (metas-rc) [reply backend/de.metas.banking/de.metas.banking.base/src/main/java/de/metas/banking/payment/impl/PaySelectionUpdater.java:389]: Done

File: backend/de.metas.payment.sepa/base/src/main/sql/postgresql/ddl/views/C_Invoice_PaySelectionTrxType_V.sql (hunk 2 above)
Comment (metas-rc) [reply backend/de.metas.payment.sepa/base/src/main/sql/postgresql/ddl/views/C_Invoice_PaySelectionTrxType_V.sql:7]: Done

**Pull request merged**